libs/libc/misc/lib_mutex.c

```diff
@@ -300,14 +322,19 @@ int nxmutex_unlock(FAR mutex_t *mutex)
 {
   int ret;
 
+  if (nxmutex_is_reset(mutex))
+    {
+      return OK;
+    }
+
   DEBUGASSERT(nxmutex_is_hold(mutex));
 
   mutex->holder = NXMUTEX_NO_HOLDER;
 
   ret = _SEM_POST(&mutex->sem);
   if (ret < 0)
     {
-      return _SEM_ERRVAL(ret);
```

Comment:
should we restore the hold:
mutex->holder = _SCHED_GETTID();

Reply:
Done